### PR TITLE
Feature/jest and working typescript

### PR DESCRIPTION
**Pull request by AdrieanKhisbe**
Builds on #3  and superseed it.

Rebased the jest features of @slinkardbrandon, migrate to codecov,
and mainly take care to have the typescript compile!! (thought rather weak typing)

This is a first setp toward good typing of vorpal.

Note I also droped gulp that was not necessary, and replace with the typescript compiler.
(`watch` and `compile`step)

### CODE REVIEW

**Review by slinkardbrandon — APPROVED** [review]

**Review by hongaar — APPROVED** [review]
🎉 let's merge!

**Comment by AdrieanKhisbe** [comment]
@hongaar !
Hell YEah 🙌 
(will try to do so this afternoon)

**Review by Ore4444 — APPROVED** [review]
wow nice! 
I love the code style consistency.

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,19 @@
 language: node_js
+sudo: false
 node_js:
   - "6"
   - "8"
   - "10"
-sudo: false
+  - "11"
```
Comment (Berkmann18) [inline .travis.yml:7]: Perhaps using `lts/*` and/or `stable` would be better than manually specifying the version (which could be subject to change in the future) would be a better idea no?

File: package.json [hunk 2]
```diff
@@ -36,53 +23,68 @@
   ],
   "author": "dthree",
   "license": "MIT",
+  "scripts": {
+    "compile": "tsc -p ./",
+    "watch": "tsc -watch -p ./",
+    "lint:base": "tslint -p tsconfig.json",
```
Comment (Berkmann18) [inline package.json:29]: FYI, `tslint` is going to be deprecated soon ("sometime in 2019").
/ref https://github.com/palantir/tslint#tslint

Comment (AdrieanKhisbe) [reply package.json:29]: ack

File: .travis.yml (hunk 1 above)
Comment (AdrieanKhisbe) [reply .travis.yml:7]: really not sure about this.

File: package.json (hunk 2 above)
Comment (Berkmann18) [reply package.json:29]: ?

File: .travis.yml (hunk 1 above)
Comment (Berkmann18) [reply .travis.yml:7]: Have a look at [this repo](https://travis-ci.org/Berkmann18/TemplateJS/) which shows how you could get your code to be tested on the latest LTS version and stable version of NodeJS regardless of the year.

File: package.json (hunk 2 above)
Comment (AdrieanKhisbe) [reply package.json:29]: acknowledged :)

File: .travis.yml (hunk 1 above)
Comment (AdrieanKhisbe) [reply .travis.yml:7]: I know it's possibile, what I mean if I doubt this is a good thing to do

File: package.json (hunk 2 above)
Comment (Berkmann18) [reply package.json:29]: Ooh, that make sense :).

File: .travis.yml (hunk 1 above)
Comment (Berkmann18) [reply .travis.yml:7]: Fair enough.

**Pull request merged**